subsys/fs/nvs/nvs.c

```diff
@@ -831,12 +832,28 @@ ssize_t nvs_read_hist(struct nvs_fs *fs, u16_t id, void *data, size_t len,
 		goto err;
 	}
 
-	return wlk_ate.len;
+	ret = wlk_ate.len;
+
+	if (more != NULL) {
+		(void) _nvs_prev_ate(fs, &wlk_addr, &wlk_ate); //save to ignore?
```

Comment:
Obviously the comment needs to go. Any Idea how best to handle an error on '_nvs_prev_ate'?

Reply:
This should be in the loop (as above) as you want to check existence of record of the known ID.
error shouldn't happened as ate was read in the last.

Error shouldn't happen - but pleas serve it as it is served above.

Reply:
@baw-serafin 
>This should be in the loop ...

Was you intention to check existence of older entry of the sane ID, or any older entry?

Reply:
Should be fixed now also rebased to curent master to include the style fixes